struct_test.go

```diff
@@ -439,3 +439,223 @@ func TestRegisterFunc_structArgs(t *testing.T) {
+		expected := ThreeDoubles{1, 2, 3}
+		if ret := ReturnThreeDoubles(1, 2, 3); ret != expected {
+			t.Fatalf("ReturnThreeDoubles returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type FourDoubles struct{ a, b, c, d float64 }
+		var ReturnFourDoubles func(a, b, c, d float64) FourDoubles
+		purego.RegisterLibFunc(&ReturnFourDoubles, lib, "ReturnFourDoubles")
+		expected := FourDoubles{1, 2, 3, 4}
+		if ret := ReturnFourDoubles(1, 2, 3, 4); ret != expected {
+			t.Fatalf("ReturnFourDoubles returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type FiveDoubles struct{ a, b, c, d, e float64 }
+		var ReturnFiveDoubles func(a, b, c, d, e float64) FiveDoubles
+		purego.RegisterLibFunc(&ReturnFiveDoubles, lib, "ReturnFiveDoubles")
+		expected := FiveDoubles{1, 2, 3, 4, 5}
+		if ret := ReturnFiveDoubles(1, 2, 3, 4, 5); ret != expected {
+			t.Fatalf("ReturnFiveDoubles returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type OneFloatOneDouble struct {
+			a float32
+			_ float32
+			b float64
+		}
+		var ReturnOneFloatOneDouble func(a float32, b float64) OneFloatOneDouble
+		purego.RegisterLibFunc(&ReturnOneFloatOneDouble, lib, "ReturnOneFloatOneDouble")
+		expected := OneFloatOneDouble{a: 1, b: 2}
+		if ret := ReturnOneFloatOneDouble(1, 2); ret != expected {
+			t.Fatalf("ReturnOneFloatOneDouble returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type OneDoubleOneFloat struct {
+			a float64
+			b float32
+		}
+		var ReturnOneDoubleOneFloat func(a float64, b float32) OneDoubleOneFloat
+		purego.RegisterLibFunc(&ReturnOneDoubleOneFloat, lib, "ReturnOneDoubleOneFloat")
+		expected := OneDoubleOneFloat{1, 2}
+		if ret := ReturnOneDoubleOneFloat(1, 2); ret != expected {
+			t.Fatalf("ReturnOneDoubleOneFloat returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type Unaligned1 struct {
+			a int8
+			_ [1]int8
+			b int16
+			_ [1]int32
+			c int64
+		}
+		var ReturnUnaligned1 func(a int8, b int16, c int64) Unaligned1
+		purego.RegisterLibFunc(&ReturnUnaligned1, lib, "ReturnUnaligned1")
+		expected := Unaligned1{a: 1, b: 2, c: 3}
+		if ret := ReturnUnaligned1(1, 2, 3); ret != expected {
+			t.Fatalf("ReturnUnaligned1 returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type Mixed1 struct {
+			a float32
+			b int32
+		}
+		var ReturnMixed1 func(a float32, b int32) Mixed1
+		purego.RegisterLibFunc(&ReturnMixed1, lib, "ReturnMixed1")
+		expected := Mixed1{1, 2}
+		if ret := ReturnMixed1(1, 2); ret != expected {
+			t.Fatalf("ReturnMixed1 returned %+v wanted %+v", ret, expected)
+		}
+	}
+	{
+		type Mixed2 struct {
+			a float32
+			b int32
+			c float32
+			d int32
+		}
+		var ReturnMixed2 func(a float32, b int32, c float32, d int32) Mixed2
+		purego.RegisterLibFunc(&ReturnMixed2, lib, "ReturnMixed2")
+		expected := Mixed2{1, 2, 3, 4}
+		if ret := ReturnMixed2(1, 2, 3, 4); ret != expected {
+			t.Fatalf("ReturnMixed2 returned %+v wanted %+v", ret, expected)
+		}
+	}
```

Comment:
Can we test a struct including all the types: float32, float64, and others like int32?

Reply:
I added one. Not sure how helpful it is since any struct bigger than 16 bytes is allocated in the Go heap and passed in for the C function to use.

Reply:
Wouldn't this be <= 16 bytes?

```go
type Mixed struct {
    a float32
    b int32
    c float64
}
```

Reply:
Yeah! Do that instead of Mixed3 or in addition to?

Reply:
Doing it instead of Mixed3 seems enough